src/main/java/reposense/parser/RepoConfigCsvParser.java

```diff
@@ -78,9 +80,18 @@ protected void processLine(List<RepoConfiguration> results, CSVRecord record) th
                     "Ignoring unknown value " + ignoreStandaloneConfig + " in ignore standalone config column.");
         }
 
+        String shallowCloningConfig = get(record, SHALLOW_CLONING_CONFIG_POSITION);
```

Comment:
If I understand correctly, in your current design, users cannot choose the depth of shallow cloning?

Reply:
We may extend to this shallow cloning argument in a nicer latest-n-commit command if we allow customizing the depth right? Correct me if I am wrong.

Reply:
> 
> 
> If I understand correctly, in your current design, users cannot choose the depth of shallow cloning?

Yes, that is correct. It is designed to replicate the report normally generated by RepoSense, so it just pulls the minimum amount of history possible while still capturing all commits between the "since" and "until" date.

> 
> 
> We may extend to this shallow cloning argument in a nicer latest-n-commit command if we allow customizing the depth right? Correct me if I am wrong.

I feel that this would be a separate feature altogether. This PR is aimed at speeding up the cloning process for large repos by reducing the size of the cloned repo. From the user's POV, the generated report would be identical (well, besides the incompatibility with `--last-modified-date`).

If I understand you correctly, the last-n-commit command would generate a report containing info about the last N commits only, i.e. it would be an alternative to the "since" and "until" dates. We can implement this for the commits report by modifying `GitLog.getWithFiles()` to look at the last N commits rather than the date range. As for the authorship report, we can implement this by just using the N-th latest commit for the `lastCommitHash`.

This means we wouldn't need to use shallow cloning, meaning we can implement the last-n-commits feature while maintaining compatibility with the `--last-modified-date` flag.

Reply:
Sorry, my bad, I previously thought you will only clone depth 1 only for analyze. It is fine in such a case.